src/Cache/QueryCacheProfile.php

```diff
@@ -26,22 +32,41 @@ class QueryCacheProfile
     private $cacheKey;
 
     /**
-     * @param int         $lifetime
-     * @param string|null $cacheKey
+     * @param int                               $lifetime
+     * @param string|null                       $cacheKey
+     * @param CacheItemPoolInterface|Cache|null $resultCache
      */
-    public function __construct($lifetime = 0, $cacheKey = null, ?Cache $resultCache = null)
+    public function __construct($lifetime = 0, $cacheKey = null, ?object $resultCache = null)
+    {
+        $this->lifetime = $lifetime;
+        $this->cacheKey = $cacheKey;
+        if ($resultCache instanceof CacheItemPoolInterface) {
+            $this->resultCache = $resultCache;
+        } elseif ($resultCache instanceof Cache) {
+            $this->resultCache = CacheAdapter::wrap($resultCache);
```

Comment:
@derrabus my team just stumbled upon this, this is a BC Break I'm afraid, if the cache key contains reserved characters.

Reply:
Can you elaborate? Or better, can you open an issue so we can discuss the BC break an possible solutions?

Reply:
This might be fixed by using `rawurlencode` instead of throwing here: https://github.com/symfony/symfony/blob/0ac7aa230fe0177214415ee20fec113a44a261f1/src/Symfony/Component/Cache/DoctrineProvider.php#L60, just like in https://github.com/symfony/symfony/blob/0ac7aa230fe0177214415ee20fec113a44a261f1/src/Symfony/Component/Cache/DoctrineProvider.php#L60

Reply:
@derrabus sorry, didn't notice your message. I will open an issue.

Reply:
See https://github.com/doctrine/dbal/issues/5051